docs: note about Vite's native SSR API being a low-level API (#3467)
vitejs/vite: docs/guide/ssr.md

The following guide also assumes prior experience working with SSR in your framework of choice, and will only focus on Vite-specific integration details.
:::

:::warning Low-level API

> This is a low-level API meant for library and framework authors. If your goal is to create an application, you may want to have a look at these higher-level SSR tools at [Awesome Vite SSR](https://github.com/vitejs/awesome-vite#ssr) first. That said, many applications are successfully built directly on top of Vite's native low-level API &mdash; just make sure to check out these higher-level tools first.

Reviewer: I think the last call to action is a bit redundant. Maybe we can do the first call to action stronger so it isn't needed:
```suggestion
This is a low-level API meant for library and framework authors. If your goal is to create an application, make sure to check out the higher-level SSR plugins and tools at [Awesome Vite SSR section](https://github.com/vitejs/awesome-vite#ssr) first. That said, many applications are successfully built directly on top of Vite's native low-level API.
```
